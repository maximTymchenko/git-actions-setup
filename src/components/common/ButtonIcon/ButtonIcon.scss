@import '../../../styles/vars';

.button__icon {
    width: 16px;
    height: 16px;

    &--secondary {
        width: 24px;
        height: 24px;
    }
}
