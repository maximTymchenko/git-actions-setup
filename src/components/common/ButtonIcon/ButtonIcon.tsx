import React, { FC } from 'react';
import './ButtonIcon.scss';

interface IButtonIconProps {
    icon?: any;
    desc?: string;
    type: string;
}

const ButtonIcon: FC = (props: IButtonIconProps) => {
    return (
        <img
            data-testid="button__icon"
            src={props.icon}
            className={`button__icon button__icon--${props.type}`}
            alt={props.desc}
        />
    );
};

export default ButtonIcon;
