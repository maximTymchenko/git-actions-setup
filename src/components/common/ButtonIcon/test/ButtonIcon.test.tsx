import React from 'react'
import ButtonIcon from '../ButtonIcon'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import arrRight from '../../../../assets/img/arr-right.svg'

it('renders elem correctly', () => {
  const { getByTestId } = render(
    <ButtonIcon icon={arrRight} desc="Test"></ButtonIcon>
  )
  expect(getByTestId('button__icon')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer
    .create(<ButtonIcon icon={arrRight} desc="Test"></ButtonIcon>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
