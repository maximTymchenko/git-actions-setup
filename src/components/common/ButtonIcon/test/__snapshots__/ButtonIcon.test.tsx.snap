// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<img
  alt="Test"
  className="button__icon button__icon--undefined"
  data-testid="button__icon"
  src="test-file-stub"
/>
`;
